Mathlib/NumberTheory/FactorisationProperties.lean

```diff
@@ -0,0 +1,320 @@
+* `Abundant` a natural number `n` is abundant if the sum of its proper divisors are greater than
+  itself and it is greater than zero
+* `Pseudoperfect` a natural number `n` is semi-perfect if a subset of its proper divisors equals
+  itself and it is greater than zero
+* `Deficient` a natural number `n` is deficient if the sum of its proper divisors is less than
+  itself and it is greater than zero
+* `WeirdNumber` a natural number is weird if it is both abundant and not semi-perfect
+
+## Main Results
+* `divisors_eq_proper_union_self` the `Finset` that is `Nat.divisors` is equal to the union of
+  `Nat.properDivisors` and the number itself
+* `deficient_or_perfect_or_abundant` any natural number greater than zero is either deficient,
+  perfect, or abundant
+* `prime_deficient` all prime natural numbers are deficient
+* `prime_pow_deficient` any power of a prime number is deficient
+
+## Implementation Details
+* Zero is not included in any of the definitions and these definitions only apply to natural
+  numbers greater than zero.
+
+## References
+* Prielipp, Robert W. “PERFECT NUMBERS, ABUNDANT NUMBERS, AND DEFICIENT NUMBERS.”
+  The Mathematics Teacher, vol. 63, no. 8, 1970, pp. 692–96. JSTOR,
+  http://www.jstor.org/stable/27958492. Accessed 21 Feb. 2024.
+
+## Tags
+abundant deficient weird Pseudoperfect
+-/
+
+open Nat BigOperators Finset
+
+set_option maxRecDepth 1000000
+
+namespace Nat
+
+variable (n m k : ℕ)
+
+/-- `n : ℕ` is Abundant if and only if the sum of the proper divisors of `n` is greater than `n`
+  and `n` is positive. -/
+def Abundant (n : ℕ) : Prop := ∑ i in properDivisors n, i > n
+
+/--  `n : ℕ` is Deficient if and only if the sum of the proper divisors of `n` is less than `n`
+  and `n` is positive -/
+def Deficient (n : ℕ) : Prop := ∑ i in properDivisors n, i < n
+
+/-- `n : ℕ` is Pseudoperfect if and only if there exists a subset of the proper divisors of n such
+  that the sum of that subset is equal to `n` and `n` is positive -/
+def Pseudoperfect (n : ℕ) : Prop :=
+  0 < n ∧ ∃ s : Finset ℕ, s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n
+
+/-- `n : ℕ` is a weird number if and only if it is both abundant and semi-perfect -/
+def WeirdNumber (n : ℕ) : Prop := Abundant n ∧ ¬ Pseudoperfect n
+
+theorem not_pseudoperfect_iff_forall :
+    ¬ Pseudoperfect n ↔ n = 0 ∨ ∀ (s : Finset ℕ), s ∈ powerset (properDivisors n)
+      → ∑ i in s, i ≠ n := by
+    have hn : (¬ 0 < n) ↔ n = 0 := by simp only [not_lt, nonpos_iff_eq_zero]
+    dsimp [Pseudoperfect]
+    rw [not_and_or, not_exists, hn]
+    constructor
+    · rintro (h1 | h2)
+      · tauto
+      · right
+        intro s
+        have hs : ¬(s ∈ powerset (properDivisors n) ∧ ∑ i in s, i = n) := by exact h2 s
+        rw [not_and] at hs
+        exact hs
+    · rintro (h1 | h2)
+      · tauto
+      · right
+        simp_rw [not_and]
+        exact h2
+
+theorem one_deficient : Nat.Deficient 1 := by
+  dsimp [Nat.Deficient]
```

Comment:
This is a so-called "non-terminal dsimp": the `dsimp` doesn't close your goal and it doesn't specify all lemmas to use.
We try to avoid these in mathlib, as adding a simp lemma could break this proof. (Unlikely in this case, but in other cases, that has happened.)

Can you replace this by an appropriate `dsimp only`? (In this case, it seems `dsimp only [Deficient]` would work; in general, you can write `dsimp?` to get a suggestion for the equivalent `dsimp only` statement.)

Reply:
Okay I did not know this. Thanks for letting me know

Reply:
I'll fix these all up